### ISSUE TITLE
[🐞] Client-side navigation does not work for node adapters

### ISSUE THREAD

**Issue by christian-hackyourshack** [BuilderIO/qwik#5404]
### Which component is affected?

Qwik City (routing)

### Describe the bug

I am trying to deploy my Qwik app to a node adapter in a docker container that runs behind an SSL proxy. When using `Link`s or programmatic routing (e.g. in qwik-auth), or form submission the client console outputs `Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'nodeType')` or similar messages, all pointing to things being undefined. All this works in dev.

The root cause might be an error in the deployment, but neither the error messages nor the documentation is of any help to find the root cause. I would expect, that a node adapter (same problem with all three express, fastify and node-server) behind a SSL proxy is fairly common for self-hosters and deserves a bit more support, at least the requirements for a deployment environment. On Discord I got some hints, that the routing requires the X-Forwarded-Host/Proto headers, but that did not help either.

### Reproduction

https://github.com/christian-hackyourshack/repro-qwik-link.git

### Steps to reproduce

You can find the results up at https://qwik.dooiy.org, try the links to the demos (flowers/todos) or the home link under the logo. Also check out the form at https://qwik.dooiy.org/demo/todolist. Whenever you submit the form it throws an `Uncaught (in promise)`. The source for this demo can be found in the above repo.

In the docker I simply call the following
```sh
pnpm install
pnpm build
PORT=80 node server/entry.express
```

### System Info

```shell
Docker: node:20-alpine3.18
behind a Traefik reverse proxy handling SSL
(the latter works fine for all my other routers)
```

**Comment by wmertens** [comment]
The HTML comments of your site have been stripped. This causes a failure on resuming, but it's not clear.

Probably your provider has a setting somewhere to optimize HTML, that should be off, the HTML is already optimized.

**Comment by wmertens** [comment]
BTW, if you run the todo without JS enabled, it fails because the server is configured to be on `http` and the site is running over `https`

```

**Comment by christian-hackyourshack** [comment]
Thank you for your very valuable observations, I will see where this comes from. I already saw those strange comments in my local devtools, but thought it might be some devtools thing. I don't think Traefik fiddles with the HTML, so it's probably the Cloudflare proxy. About the http vs https I have to see, where that comes from. The server actually runs only http and Traefik adds the SSL layer. That's why I set the ORIGIN. I will check tomorrow. Another chance for Qwik. I am very much amazed by the technology and would love to employ it.

**Comment by christian-hackyourshack** [comment]
I tried it out and could fix both settings: The first was due to the default HTML minification in Cloudflare, the other was just in the repro, that I put together qwikly, that I forgot to set the ORIGIN.

In order to open up Qwik to a broader audience, I would love to see and contribute to the documentation. I think a deployment page for self-hosters would be better than the statement "Since you are choosing <xyz>, here you are in your own, after running npm run build:". I think we can do better in supporting new adopters.

I will open up a discussion for "Deployment page for self-hosters".
